viewer/packages/tools/src/Geomap/Geomap.ts

```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+// To overcome "implicitly has an 'any' type" and "unused variable reference" error in the geo-three library
+// @ts-ignore
+import * as GEOTHREE from 'geo-three';
+import { Cognite3DViewer } from '@reveal/core';
+import { LatLongPosition, MapConfig, MapProviders } from './MapConfig';
+
+export class Geomap {
+  private readonly _viewer: Cognite3DViewer;
+  private _map: GEOTHREE.MapView;
+  private _intervalId: any = 0;
+
+  constructor(viewer: Cognite3DViewer, mapConfig: MapConfig) {
+    this._viewer = viewer;
+    const mapProvider = this.getMapProvider(mapConfig);
+    this._map = new GEOTHREE.MapView(GEOTHREE.MapView.PLANAR, mapProvider, mapProvider);
+    this._viewer.addObject3D(this._map);
+
+    const coords = GEOTHREE.UnitsUtils.datumsToSpherical(mapConfig.latlong.latitude, mapConfig.latlong.longitude);
+    const bound = this._viewer.models[0].getModelBoundingBox();
+    this._map.position.set(-coords.x, bound.min.y, coords.y);
+    this._map.updateMatrixWorld(true);
+    this.requestRedraw(10000);
+    this._viewer.on('cameraChange', () => {
+      this.requestRedraw(5000);
+    });
+  }
+
+  private requestRedraw(timeOut: number) {
+    if(this._intervalId == 0) {
+      this._intervalId = setInterval(() => {
+        this._viewer.requestRedraw();
+      }, 100);
+    }
+
+    setTimeout(() => {
+      clearInterval(this._intervalId);
+      this._intervalId = 0;
+    }, timeOut);
+  }
```

Comment:
I think you've misunderstood me. Imagine this:
1. cameraChanged is triggered. `this._intervalId` is set and an interval timer is started, requesting redraw every 100 ms
2. cameraChanged is triggered again, before the first interval timer is stopped. A new timer is started, and `this._intervalId` is overwritten.

After a while the `setTimeout` call from 1) is fired. `clearInterval(this._intervalId)` clears the timer from 2, but the timer from 1. *will never be cleared* - running for all eternity. 

Now add that cameraChanged will be triggered _alot_ and it's clear that things will go quite badly.

Instead, try something like: (note - not tested, so might need some adjustments)

```suggestion
  private startRequestRedrawTimer(timeout: number) {
    this.clearRequestRedrawTimer();
    
    this._intervalId = setInterval(() => {
      this._viewer.requestRedraw();
    }, 200);
    this._timeoutId = setTimeout(() => {
      this.clearRequestRedrawTimer();
    }, timeout);
  }
  
  private clearRequestRedrawTimer() {
    if (this._intervalId !== 0) {
      clearInterval(this._intervalId);
      clearTimeout(this._timeoutId);
      this._intervalId = 0;
      this._timeoutId = 0;
    }
  }
```

Reply:
@larsmoa the above existing code will have following workflow.

1. `cameraChanged` is triggered and `this._intervalId` is set and an interval timer is started, requesting redraw every 100 ms.
2. Again `cameraChanged` is triggered, but this time `this._intervalId` value is not change because `if(this._intervalId == 0)` is false and `setInterval()` is not called until previous `setInterval()` is cleared which will happen when `timeOut` duration is completed.
3. But `setTimeout()` should have been under the `if(this._intervalId == 0)` is avoid unnecessary `clearInterval()`.

Let me know if this needs any change